sqlserver/tests/test_metrics.py

```diff
@@ -0,0 +1,583 @@
+
+    for metric_name in EXPECTED_QUERY_EXECUTOR_AO_METRICS_PRIMARY:
+        for tag_prefix in (
+            'availability_group',
+            'availability_group_name',
+            'availability_mode',
+            'database_id',
+            'database_name',
+            'database_state',
+            'failover_cluster',
+            'failover_mode',
+            'replica_id',
+            'replica_role',
+            'replica_server_name',
+            'synchronization_state',
+        ):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name in EXPECTED_QUERY_EXECUTOR_AO_METRICS_REPLICA_COMMON:
+        for tag_prefix in (
+            'availability_group',
+            'availability_group_name',
+            'availability_mode',
+            'database_id',
+            'database_name',
+            'database_state',
+            'failover_cluster',
+            'failover_mode',
+            'replica_id',
+            'replica_role',
+            'replica_server_name',
+            'synchronization_state',
+        ):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name in EXPECTED_QUERY_EXECUTOR_AO_METRICS_QUORUM_COMMON:
+        for tag_prefix in (
+            'quorum_type',
+            'quorum_state',
+            'failover_cluster',
+        ):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name in EXPECTED_QUERY_EXECUTOR_AO_METRICS_MEMBER_COMMON:
+        for tag_prefix in (
+            'member_name',
+            'member_type',
+            'member_state',
+            'failover_cluster',
+        ):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name, _, _ in AO_METRICS_SECONDARY:
+        aggregator.assert_metric(metric_name, count=0)
+
+
+@not_windows_ci
+@always_on
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_check_ao_secondary_replica(aggregator, dd_run_check, init_config, instance_ao_docker_secondary):
+    sqlserver_check = SQLServer(CHECK_NAME, init_config, [instance_ao_docker_secondary])
+    dd_run_check(sqlserver_check)
+
+    for metric_name, _, _ in AO_METRICS_SECONDARY:
+        for tag_prefix in ('availability_group', 'availability_group_name', 'synchronization_health_desc'):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name, _, _ in AO_AG_SYNC_METRICS:
+        for tag_prefix in ('availability_group', 'availability_group_name', 'synchronization_health_desc'):
+            aggregator.assert_metric_has_tag_prefix(metric_name, tag_prefix=tag_prefix)
+
+    for metric_name, _, _ in AO_REPLICA_SYNC_METRICS:
+        for tag_prefix in (
+            'availability_group',
+            'availability_group_name',
+            'synchronization_state_desc',
+            'replica_server_name',
```

Comment:
I wonder if there is a way to include this set of tags in the actual `AO_REPLICA_SYNC_METRICS` definition in the conf file, so when someone searches for the metric in our code base the following info is clear:

1. The metric name
2. Tags that apply to said metric
3. The tables from which we query for this metric 
4. Possibly (???) the query itself, so people don't have to reverse engineer that metrics.py file